trivup/apps/OauthbearerOIDCApp/HttpServer.py

```diff
@@ -0,0 +1,81 @@
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import jwt
+import datetime
+
+class WebServerHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        message = "HTTP server for OAuth\n"
+        message += "Example for token retrieval:\n"
+        message += 'curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "accept: application/json" \
+        -H "authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "cache-control: no-cache" \
+        -H "content-type: application/x-www-form-urlencoded" \
+        -d "method=oidc,scope=test-scope"'
+        self.wfile.write(message.encode())
+        print (message)
+
+    def do_POST(self):
+        '''
+        Example usage:
+        curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "accept: application/json" \
+        -H "authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "cache-control: no-cache" \
+        -H "content-type: application/x-www-form-urlencoded" \
+        -d "method=oidc,scope=test-scope"
+        '''
+        if self.path.endswith("/retrieve"):
+
+            content_length = int(self.headers['Content-Length'])
+            post_data = self.rfile.read(content_length)
+            
+            if not self.headers['Content-Type']: {
```

Comment:
Are these headers guaranteed to be in self.headers? If not you need to use ` self.headers.get(.., None)`.

Also use `if .. is None` rather than `if not`, since `if not` is a bit too smart.